test/unittest/xmlparser/XMLEndpointParserTests.cpp

```diff
@@ -0,0 +1,805 @@
+                            <livelinessQos kind=\"AUTOMATIC_LIVELINESS_QOS\" leaseDuration_ms=\"1000\"/>\
+                        </reader>\
+                    </participant>\
+                    <participant>\
+                        <name>HelloWorldPublisher</name>\
+                        <writer>\
+                            <unicastLocator address=\"192.168.0.120\" port=\"9000\"/>\
+                            <unicastLocator address=\"10.47.8.31\" port=\"8000\"/>\
+                            <multicastLocator address=\"239.255.1.1\" port=\"7000\"/>\
+                            <userId>5</userId>\
+                            <entityID>6</entityID>\
+                            <topicName>HelloWorldTopic</topicName>\
+                            <topicDataType>HelloWorld</topicDataType>\
+                            <topicKind>WITH_KEY</topicKind>\
+                            <partitionQos>HelloPartition</partitionQos>\
+                            <partitionQos>WorldPartition</partitionQos>\
+                            <reliabilityQos>BEST_EFFORT_RELIABILITY_QOS</reliabilityQos>\
+                            <durabilityQos>VOLATILE_DURABILITY_QOS</durabilityQos>\
+                            <ownershipQos kind=\"SHARED_OWNERSHIP_QOS\" strength=\"50\"/>\
+                            <livelinessQos kind=\"AUTOMATIC_LIVELINESS_QOS\" leaseDuration_ms=\"1000\"/>\
+                        </writer>\
+                    </participant>\
+                </staticdiscovery>\
+                ";
+        ASSERT_EQ(tinyxml2::XMLError::XML_SUCCESS, xml_doc.Parse(xml));
+        EXPECT_EQ(XMLP_ret::XML_OK, mp_edpXML->loadXMLNode(xml_doc));
+    }
+
+    {
+        // Wrong XML
+        const char* xml ="<bad_xml></bad_xml>";
+        ASSERT_EQ(tinyxml2::XMLError::XML_SUCCESS, xml_doc.Parse(xml));
+        EXPECT_EQ(XMLP_ret::XML_ERROR, mp_edpXML->loadXMLNode(xml_doc));
+    }
+
+}
+
+
+
+
+/*
+ * This test checks the XMLEndpointParser::loadXMLReaderEndpoint method.
+ * 2. Check incorrect values for the writer tag
+ * 2. Check incorrect values for the reader tag
+ * 3. Check an incorrect tag
+ */
+TEST_F(XMLEndpointParserTests, loadXMLParticipantEndpoint)
+{
+    tinyxml2::XMLDocument xml_doc;
+    tinyxml2::XMLElement* titleElement;
+
+    mock_consumer = new eprosima::fastdds::dds::XMLMockConsumer(xml_mutex_);
+
+    Log::RegisterConsumer(std::unique_ptr<LogConsumer>(mock_consumer));
+    Log::SetVerbosity(Log::Warning);
+
+    {
+        StaticRTPSParticipantInfo* pdata = new StaticRTPSParticipantInfo();
+        // Parametrized XML
+        const char* xml =
+                "\
+                <participant>\
+                    <name>HelloWorldSubscriber</name>\
+                    <reader>\
+                        <userId>3</userId>\
+                        <entityID>4</entityID>\
+                        <expectsInlineQos>true</expectsInlineQos>\
+                        <topicName>HelloWorldTopic</topicName>\
```

Comment:
Remember to include here the `<topic>` element as it won't be deleted in the `XMLEndpointParser.cpp` file